include/cute/atom/copy_traits_xe.hpp

```diff
@@ -2022,4 +2068,103 @@ namespace detail
   }
 } // end namespace detail
 
+template <class TiledCopy, class ThrIdx>
+class Xe2DThrCopy : ThrCopy<TiledCopy, ThrIdx> {
+
+public:
+
+  CUTE_HOST_DEVICE
+  Xe2DThrCopy(ThrIdx const& thr_idx) : ThrCopy<TiledCopy, ThrIdx> (thr_idx) {}
+
+  template <class DTensor>
+  CUTE_HOST_DEVICE
+  auto
+  retile_D(DTensor&& dtensor) {
+    if constexpr (!TiledCopy::is_convention_MN) {
+      return retile_D_nkl(dtensor);
+    } else {
+      return retile_D_mkl(dtensor);
+    }
+  }
+
+private:
+
+  template <class DTensor>
+  CUTE_HOST_DEVICE static
+  auto
+  retile_D_mkl(DTensor&& dtensor) {
+    auto tmp = ThrCopy<TiledCopy, ThrIdx>::retile_D(dtensor);
+    return make_tensor(static_cast<decltype(tmp) &&>(tmp).data(),
+                           tmp.shape());
+  }
+
+  template <class DTensor>
+  CUTE_HOST_DEVICE static
+  auto
+  retile_D_nkl(DTensor&& dtensor) {
+    auto b_tensor = make_tensor(dtensor.data(),
```

Comment:
why can't we instead swap the tiler dimensions? That way we we shouldn't need to distinguish between mkl/nkl. And we wouldn't need Xe2DThrCopy.